### ISSUE TITLE
Transformations in sorted lists

### ISSUE THREAD

**Issue by flsmith** [gap-packages/Semigroups#251]
Bug occurs on GAP 4.8.6 with Semigroups unstable-3.0
The following code shows that a number of transformations that are in a sorted list are not found in the list:

```
gap> S := Monoid([
> Transformation([2, 1]),
> Transformation([3, 1, 2]),
> Transformation([4, 2, 3, 4]),
> Transformation([2, 2])]);;
gap> slist := ShallowCopy(AsList(S));;
gap> Sort(slist);
gap> r := MultiplicativeNeutralElement(S);;
gap> for s in S do
> if not r * s in slist then
> Print(s);
> fi;
> od;
```

Proper behaviour should be that nothing is printed. @wilfwilson has confirmed that with Semigroups loaded, a number of elements are printed, but without the code runs as expected.
 
Without the Sort(), the code runs as expected.

**Comment by wilfwilson** [comment]
And I'll note that without the Semigroups package loads, this bug does not appear. So it does seem to be an issue with Semigroups.

**Comment by flsmith** [comment]
Does not occur in Semigroups 2.8.0, however.

**Comment by ChrisJefferson** [comment]
This looks like it might be painful to debug. Let me know if you get stuck.

I'd try pushing something onto `slist`, to find out if it is the sorting, or `in` on a fully sorted list is the problem.

**Comment by james-d-mitchell** [comment]
I think I know what this is, and I can fix it.

**Comment by james-d-mitchell** [comment]
This is a bug in the library file `src/trans.c` specifically in the function `LtTrans24`, there are two for-loops which are incorrectly nested. This wasn't picked up before because it is not possible to create transformations within GAP that satisfy the condition required for this code to be executed. 

It happens with Semigroups 3.0 because we create small degree `TRANS_4`s in the kernel module. I'll push a fix to GAP in a couple of minutes resolving this.

**Comment by james-d-mitchell** [comment]
Resolved by [GAP PR#1130](https://github.com/gap-system/gap/pull/1130)

**Comment by james-d-mitchell** [comment]
But we should add Finn's example to `testinstall.tst`.
